.travis.yml

```diff
@@ -428,3 +428,175 @@ script:
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+#
+# Disabled due to compile failure on E0_AUTO_FAN_PIN PB8
+#
+#  - use_example_configs Malyan/M200
+#  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+#  - restore_configs
+  - use_example_configs Micromake/C1/basic
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs Micromake/C1/enhanced
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs RepRapWorld/Megatronics
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs RigidBot
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs SCARA
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs Velleman/K8200
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs Velleman/K8400/Dual-head
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs Velleman/K8400
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+#
+# Disabled due to compile failure on https://github.com/stawel/SlowSoftI2CMaster
+#
+#  - Requires https://github.com/stawel/SlowSoftI2CMaster
+#  - restore_configs
+#  - use_example_configs wt150
+#  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+
+
+  - export TEST_PLATFORM="-e rambo"
+  - restore_configs
+  - use_example_configs AlephObjects/TAZ4  - restore_configs
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+
+
+  - export TEST_PLATFORM="-e LPC1768"
+  - restore_configs
+  - use_example_configs FolgerTech/i3-2020
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs Mks/Sbase
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs Wanhao/Duplicator 6
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+
+
+  - export TEST_PLATFORM="-e anet10"
+  - restore_configs
+  - use_example_configs Anet/A6
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs Anet/A8
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs Creality/CR-10
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs Malyan/M150
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs Sanguinololu
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+  - restore_configs
+  - use_example_configs TinyBoy2
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
+
+
+  - export TEST_PLATFORM="-e sanguino_atmega644p"
+  - restore_configs
+  - use_example_configs tvrrug/Round2
+  - build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
```

Comment:
I can't even begin to imagine how long it takes to do all these builds. There's simply no way we can have Travis CI doing every single one of these for every single HEAD commit. This severely slows the development cycle, as it will now take 30 minutes to get feedback on changes.

Reply:
Agreed that it isn't a good long-term solution but the fact is that with the "fast and loose" method that Marlin is deloped today, I know that I am not the only one who tires of the endless regressions that are "found" by users rather than Travis.  I'm sure you've see the complaints by those (usually with incorrigible attitudes, but  with very valid points nonetheless) who find the wild-wild-west of the marlin project too much to swallow.

I suggest you merge and then begin looking at ways to resolve (either via nightly "comprehensive" test - which will likely go largely ingored and have no utility) or parallelize the targets.  Rather than do either, I was going to try and tackle a more comprehensive solution that would do away with the current platformio.ini structure and have a single "Marlin 2.0.x" target that would pickup the PIO stuff for the target from the Configuration, then the targets could be tested in parallel with a simple "targetsA", "targetsB", "targetsC",... etc config.

But yes, don't have any of that done.  All that seems to be complete is a very fast way to continue breaking builds with every passing commit :)

Reply:
A good point that came up was the need to depricate examples.  Kossel Pro is from 2013 (and failing to build) so suggest we move to "Unmaintained" dir within Configs.  Are there more in these configs that should go ?  The most valuable Travis checks are for printers that people are actually using.

But what is the upper limit on Travis completion ?  If we start there, it will be easier to determine how to hit it with the least effort.

Reply:
I noticed this in the travis config:

``` 
  ######## Board Types #############
  #
  # To be added in nightly test branch
  #
```

Is there such a thing ?

Reply:
More Travis CI testing is not a comprehensive solution. It would be better to produce test configs (`config/tests`) with many, many options enabled, and just build those 2 or 3 configs with each MCU.

Reply:
> Kossel Pro is from 2013 (and failing to build) so suggest we move to "Unmaintained" dir within Configs. 

Or just patch it up so it builds. It's "maintained" if someone would simply patch it when an error is found. For the most part, very few of the example configs have a dedicated maintainer.

Reply:
> # To be added in nightly test branch

Wishful thinking on the part of the commenter. In reality there are no contributors who can be available every day to make some kind of "nightly build", collate the TODOs, markup all the milestones, etc. The `bugfix-x.x.x` branches are as "nightly" as we get.

Reply:
"Or just patch it up so it builds."

Go for it ;) -  The original submitter and I had already discussed in #8795 without a good solution.

"More Travis CI testing is not a comprehensive solution. It would be better to produce test configs (config/tests) with many, many options enabled, and just build those 2 or 3 configs with each MCU."

Stating the obvious doesn't resolve the issue :).  This was meant to be a stop-gap until something better is done.  Choosing known-working-at-one-time configs seemed better than nothing and did not take much effort (plus improves another aspect, that of the example configs actually working).  The fact that the example configs caught things is *prima facie* proof that they catch regressions that are not being caught today.  I'm apparently a masochist because I persevered with Marlin for nearly 6 months with a broken Printrboard build early this year since it was 2 steps forward, 2 steps back through 3 complete 1.x releases.   As the 32-bit targets continue to come online, it will likely get really <strike>bad</strike> worse long before it gets any better so continuing to let the status quo be the best that can be done seems pretty lackluster of a proposition.

"Wishful thinking on the part of the commenter."

So no nightly instance of more comprehensive regression testing, and no realtime improvement in regression testing.  That's somewhat discouraging 😞 .